automation/deploy-bin.ts

```diff
@@ -15,70 +15,131 @@
  * limitations under the License.
  */
 import * as archiver from 'archiver';
-import * as Promise from 'bluebird';
+import * as Bluebird from 'bluebird';
 import * as fs from 'fs-extra';
+import * as _ from 'lodash';
 import * as mkdirp from 'mkdirp';
-import * as os from 'os';
 import * as path from 'path';
 import * as publishRelease from 'publish-release';
 
-import * as packageJSON from '../package.json';
-
-const publishReleaseAsync = Promise.promisify(publishRelease);
-const mkdirpAsync = Promise.promisify<string | null, string>(mkdirp);
-
+const mkdirpAsync = Bluebird.promisify<string | null, string>(mkdirp);
 const { GITHUB_TOKEN } = process.env;
 const ROOT = path.join(__dirname, '..');
-
+// Note: the following 'tslint disable' line was only required to
+// satisfy ts-node under Appveyor's MSYS2 on Windows -- oddly specific.
+// Maybe something to do with '/' vs '\' in paths in some tslint file.
+// tslint:disable-next-line:no-var-requires
+const packageJSON = require(path.join(ROOT, 'package.json'));
 const version = 'v' + packageJSON.version;
-const outputFile = path.join(
-	ROOT,
-	'build-zip',
-	`balena-cli-${version}-${os.platform()}-${os.arch()}.zip`,
-);
+const arch = process.arch;
+
+function dPath(...paths: string[]) {
+	return path.join(ROOT, 'dist', ...paths);
+}
+
+const standaloneZips: { [platform: string]: string } = {
```

Comment:
I know they don't explicitly get used anywhere, but given this and the next three all end up as the same essential keyed type, might be nice to explicitly define that type.